mandible/metadata_mapper/format/h5.py

```diff
@@ -18,9 +18,38 @@ def parse_data(file: IO[bytes]) -> contextlib.AbstractContextManager[Any]:
 
     @staticmethod
     def eval_key(data: Any, key: Key) -> Any:
+        parsed_key, parsed_attribute = parse_key(key.key)
+        if parsed_attribute:
+            return normalize(data[parsed_key].attrs.get(parsed_attribute))
         return normalize(data[key.key][()])
```

Comment:
I'd maybe write it like this. Important thing I just noticed while doing this, we should not be using `data[key.key]` anymore, we should be using `data[group_key]` since `key.key` will possibly contain escape characters.

You'll probably see this come up as an issue if you add a test where the h5 group contains an `@`.
```suggestion
        group_key, attr_name = parse_key(key.key)
        if attr_name is not None:
            val = data[group_key].attrs.get(attr_name)
        else:
            val = data[group_key][()]

        return normalize(val)
```

The thing I like about this way is the lines are a bit shorter and there's only one call to `normalize` although there is an `else` which I like to avoid when possible.

Reply:
I added the case and changed the grouping but I think I still prefer the two returns over 1.